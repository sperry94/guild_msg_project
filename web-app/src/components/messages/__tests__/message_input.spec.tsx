import React from "react"
import renderer from "react-test-renderer"
import MessageInput, { MessageInputProps } from "../message_input"

describe("MessageInput", () => {
  const cases: MessageInputProps[] = [
    {
      conversationId: undefined,
      disabled: false,
    },
    {
      conversationId: "convo-id",
      disabled: false,
    },
    {
      conversationId: undefined,
      disabled: true,
    },
    {
      conversationId: "convo-id",
      disabled: true,
    },
  ]

  cases.forEach((c, index) => {
    it(`renders for case #${index + 1}`, () => {
      const tree = renderer.create(<MessageInput {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
