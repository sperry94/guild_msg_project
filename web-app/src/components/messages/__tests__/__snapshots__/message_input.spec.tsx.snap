// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput renders for case #1 1`] = `
<div
  className="MuiPaper-root makeStyles-content-1 MuiPaper-elevation5"
>
  <form
    action="#"
    className="makeStyles-form-2"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          aria-label="Message"
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          name="message"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-73 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-74"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Send
      </span>
    </button>
  </form>
</div>
`;

exports[`MessageInput renders for case #2 1`] = `
<div
  className="MuiPaper-root makeStyles-content-1 MuiPaper-elevation5"
>
  <form
    action="#"
    className="makeStyles-form-2"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          aria-label="Message"
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          name="message"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-73 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-74"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Send
      </span>
    </button>
  </form>
</div>
`;

exports[`MessageInput renders for case #3 1`] = `
<div
  className="MuiPaper-root makeStyles-content-1 MuiPaper-elevation5"
>
  <form
    action="#"
    className="makeStyles-form-2"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          aria-label="Message"
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
          disabled={true}
          name="message"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-73 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-74"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Send
      </span>
    </button>
  </form>
</div>
`;

exports[`MessageInput renders for case #4 1`] = `
<div
  className="MuiPaper-root makeStyles-content-1 MuiPaper-elevation5"
>
  <form
    action="#"
    className="makeStyles-form-2"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          aria-label="Message"
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
          disabled={true}
          name="message"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-73 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-74"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Send
      </span>
    </button>
  </form>
</div>
`;
